### PR TITLE
Updated to Slick 3.1 best practices

### PR DESCRIPTION
**Pull request by jmcardon**
Hi @pedrorijo91. I used this template almost a year ago when I was learning slick and it helped me lots.

Now that slick 3.1 has evolved, I wanted to contribute to update to the play-slick 3.1 best practices, since there's no need to inject a Provider[Application] when extending the trait HasDatabaseConfigProvider[JdbcProfile] and injecting an instance of DatabaseConfigProvider. The trait takes care of that for you.

### CODE REVIEW

File: app/models/User.scala [hunk 1]
```diff
@@ -37,28 +39,29 @@ class UserTableDef(tag: Tag) extends Table[User](tag, "user") {
     (id, firstName, lastName, mobile, email) <>(User.tupled, User.unapply)
 }
 
-object Users {
+@Singleton
+class Users @Inject() (protected val dbConfigProvider: DatabaseConfigProvider) extends HasDatabaseConfigProvider[JdbcProfile] {
 
-  val dbConfig = DatabaseConfigProvider.get[JdbcProfile](Play.current)
+  import driver.api._
```
[the remark this answers is not in the document]
Comment (jmcardon) [reply app/models/User.scala:45]: This import can only be done inside of the class, as driver is an inherited member of HasDatabaseConfigProvider

**Comment by pedrorijo91** [comment]
Oh sorry, I remember to have seen this email, but somehow it didn't go to my TODO list and I completely forgot about it :/

I'm really happy this template could help someone!

It seems great on the overall. Just a couple of questions: 

- This PR is meant to update to play 2.5.x and slick 2.x right? 
- Wouldn't it make sense to update to latest 2.5.x play version (2.5.13 i think) instead of 2.5.4?
- Same for play-slick. why 2.0.0 instead of 2.1.0? (I'm not aware of any major change between the versions you choose and the latest at least, so it would be straightforward to update the versions)
- Have you seen there's already a play 2.5.x branch at https://github.com/pedrorijo91/play-slick3-steps/tree/play2.5 ? If so, would you like to compare your changes with that and see if you think it's worth to do the changes you find relevant on top of that branch?

Thanks a lot for the contribution!

File: app/models/User.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedrorijo91) [reply app/models/User.scala:45]: yeah, just ignore this (and the other comments). I was thinking about disable them for a while

**Comment by jmcardon** [comment]
@pedrorijo91 oh crap my bad. 

Yeah, I meant to PR the 2.5 branch, not the other one.

I'll close this pr and make another one.

edit: I did fork this from your branch. Let me try to fix the versions to the latest.

**Comment by jmcardon** [comment]
As far as why 2.0.2 as opposed to 2.1.0, mostly because AFAIK 2.0.2 is using 3.1 and is stable. I haven't heard much word from the play-slick guys when it comes to the two versions, but I could ask on gitter. It seems 2.1.0 is still an active branch that is being updated.

**Comment by pedrorijo91** [comment]
fair enough :)

**Pull request closed without merging**